--- gtags.conf.orig	2011-08-16 14:54:40.000000000 -0400
+++ gtags.conf	2011-08-16 14:57:20.708588879 -0400
@@ -33,7 +33,7 @@ common:\
 	:skip=HTML/,HTML.pub/,tags,TAGS,ID,y.tab.c,y.tab.h,cscope.out,cscope.po.out,cscope.in.out,SCCS/,RCS/,CVS/,CVSROOT/,{arch}/,autom4te.cache/:
 gtags:\
 	:tc=common:\
-	:langmap=c\:.c.h,yacc\:.y,asm\:.s.S,java\:.java,cpp\:.c++.cc.cpp.cxx.hxx.hpp.C.H,php\:.php.php3.phtml:
+	:langmap=c\:.c.h,yacc\:.y,asm\:.s.S,java\:.java,cpp\:.c++.cc.cpp.cxx.hxx.hpp.C.h.H,php\:.php.php3.phtml:
 #
 # plug-in parser (for Exuberant Ctags 5.8)
 #
