--- etckeeper.conf.orig	2012-08-12 23:15:34.613756253 -0400
+++ etckeeper.conf	2012-08-12 23:17:28.395718519 -0400
@@ -31,8 +31,8 @@
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum, zypper etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+HIGHLEVEL_PACKAGE_MANAGER=pacman
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+LOWLEVEL_PACKAGE_MANAGER=pacman
