--- etckeeper.conf.orig	2011-07-22 14:22:24.463411702 -0400
+++ etckeeper.conf	2011-07-22 14:24:19.733414752 -0400
@@ -27,8 +27,8 @@ DARCS_COMMIT_OPTIONS="-a"
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+HIGHLEVEL_PACKAGE_MANAGER=pacman
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+LOWLEVEL_PACKAGE_MANAGER=pacman
